src/javascripts/Factories/contextMenuFactory.js

```diff
@@ -161,6 +161,33 @@ ngapp.service('contextMenuFactory', function() {
             });
         }
     }, {
+        id: 'Masters',
+        visible: (scope) => {
+            if (!scope.selectedNodes.length) return;
+            let nodes = scope.selectedNodes, node = nodes.last();
+            return node.element_type === xelib.etFile && xelib.GetIsEditable(node.handle);
+        },
+        build: (scope, items) => {
+            let node = scope.selectedNodes.last(),
+                children = [];
+            children.push({
```

Comment:
why define children as an empty array then push an array of stuff onto it?  Why not just define the children array with this stuff already in it?  or, heck, not have a children array at all?

are you thinking ahead to some future revision of the code where some of these items would not be displayed?  I don't think that will ever be the case here here, I think we're likely to have all these items visible whenever the `Masters` submenu is available (although some may be disabled in certain circumstances).